### DISCUSSION TITLE
Build caching

### DISCUSSION THREAD

**Discussion by wmertens** [QwikDev/qwik-evolution#187] [Proposals For Qwik]
### What is it about?

make the optimizer cache build results

#### Problems you are trying to solve:

While the optimizer is fast, not running it is even faster.

---

#### What do you propose?

In the transform phase of the plugin, we could hash the code+transform options and then check a cache under `node_modules/.qwik`. This is especially nice in dev mode.

I'm not sure what vite or rollup already offer to assist with this.
We must also make sure the plugin can still work without filesystem access.

---

### Links / References

https://github.com/vitejs/vite/discussions/12943

**Comment by shairez** [comment]
Thanks @wmertens 
We went over this and thought it's a good idea to explore

**Comment by wmertens** [comment]
On further reflection it would add a file system dependency to the optimizer for not that much gain, I think the correct place for this is inside vite.
